ui/src/treeDataProvider/AzureAccountTreeItemBase.ts

```diff
@@ -61,49 +62,49 @@ export abstract class AzureAccountTreeItemBase extends AzExtParentTreeItem imple
     }
 
     public async loadMoreChildrenImpl(_clearCache: boolean, context: types.IActionContext): Promise<AzExtTreeItem[]> {
-        let azureAccount: AzureAccount | undefined = await this._azureAccountTask;
-        if (!azureAccount) {
+        this.azureAccount = await this._azureAccountTask;
```

Comment:
> Expose an AzureAccount property so that children can easily check if the user is signed in, etc.

Please spend a little bit more time on your PR descriptions, especially in this repo because it affects many extensions. We can usually figure out _what_ you're doing by looking at the changes (although a short description doesn't hurt), but I want to know more about _why_ you're doing something. For example, why do children need to easily check if the user is signed in?

Reply:
Apologies for the lack of description.

In the Azure App Service extension we end up checking if a user is signed in to decide deploy behavior.  I didn't like how using `if (children.length > 0 && children[0] instanceof GenericTreeItem)` felt. But I see that Functions does the same thing so maybe this change is unnecessary and I can just keep checking `if (children.length > 0 && children[0] instanceof GenericTreeItem)`. @nturinski and I decided that this would be a good way of checking if a user is signed in without checking the child nodes.

Reply:
How does it affect deploy behavior if a user is signed in?

As mentioned in my other comment, I don't want to expose the "AzureAccount" types in this package. I suggest you either check `children` like functions is doing or find another way around the problem without exposing the full `AzureAccount`. For example, maybe you just add a simple "isSignedIn" property to this class. That way, you're only exposing a boolean instead of an entire class

Reply:
If the user has a trial app, but is not signed in, then we can automatically deploy to their trial app. If the user is signed in then we have to go through other steps to get the deploy target.

> For example, maybe you just add a simple "isSignedIn" property to this class. That way, you're only exposing a boolean instead of an entire class

Plan on doing this now.

Reply:
Hmm as of today, I'm guessing it would show these three options in the tree item picker when they deploy while not signed-in:
- trial app
- Sign in to azure
- Create free azure account

I agree it's a little more annoying to have to click the trial app - but isn't it important for us to highlight the other two options?

Reply:
I don't think this would be a the best user flow. Especially since the majority of trial app users won't be signed in (at least that is our goal). And for this user, we want the deploy button to do everything for the user. We might have to discuss this more with @fiveisprime.